src/Features/CSharp/Portable/ExternalAccess/Pythia/PythiaSignatureHelpProvider.cs

```diff
@@ -21,23 +24,41 @@ namespace Microsoft.CodeAnalysis.ExternalAccess.Pythia
     [ExtensionOrder(Before = "InvocationExpressionSignatureHelpProvider")]
     internal sealed class PythiaSignatureHelpProvider : InvocationExpressionSignatureHelpProviderBase
     {
-        private readonly IPythiaSignatureHelpProviderImplementation _implementation;
+        private readonly Lazy<IPythiaSignatureHelpProviderImplementation?> _lazyImplementation;
 
         [ImportingConstructor]
-        public PythiaSignatureHelpProvider(IPythiaSignatureHelpProviderImplementation implementation)
+        public PythiaSignatureHelpProvider([Import(AllowDefault = true)] Lazy<IPythiaSignatureHelpProviderImplementation?> implementation)
```

Comment:
```suggestion
        public PythiaSignatureHelpProvider(Lazy<IPythiaSignatureHelpProviderImplementation?> implementation)
```

Reply:
Why should it be required? Shouldn't Roslyn with without IntelliCode installed?

Reply:
MEF will silently omit this part from the MEF catalog when IntelliCode is not installed (or is disabled). We do not want this part available in those cases.

Reply:
In other words, `PythiaSignatureHelpProvider` is a MEF export if and only if IntelliCode is enabled.

Reply:
@KirillOsenkov says an error is reported in VS4Mac:

```
INFO [2020-02-28 11:20:58Z]: MEF composition error: Microsoft.CodeAnalysis.ExternalAccess.Pythia.PythiaSignatureHelpProvider.ctor(implementation): expected exactly 1 export matching constraints:
Contract name: Microsoft.CodeAnalysis.ExternalAccess.Pythia.Api.IPythiaSignatureHelpProviderImplementation
TypeIdentityName: Microsoft.CodeAnalysis.ExternalAccess.Pythia.Api.IPythiaSignatureHelpProviderImplementation
but found 0.
```

Should they just ignore it?

Reply:
@tmat VS for Mac is reporting an error for a by-design roslyn scenario. The workaround is to explicitly remove this part from their MEF catalog before creating the export provider, rather than relying on the automatic exclusion we are using.

Reply:
Here's a more complete explanation:

> We are using MEF part rejection as part of the Roslyn design. We have `[Export]` components with required imports, knowing that in some scenarios the import will not be available and the linked export will silently also not be part of the catalog. We are using this to simulate a behavior where a MEF export in package A behaves as though it was provided by package B, where package A is always available but package B is optional. By implementing the export in package A with a required import from package B, the export becomes fully-conditioned on B.

Reply:
I see. OK, makes sense.

Reply:
@NTaylorMullen FYI - this applies to Razor as well.